web/packages/teleport/src/Navigation/SideNavigation/Navigation.tsx

```diff
@@ -288,11 +328,39 @@ export function Navigation() {
     [debouncedSection]
   );
 
-  const resetExpandedSection = useCallback((closeAfterDelay = true) => {
-    setIsClosing(closeAfterDelay);
-    setPreviousExpandedSection(null);
-    setTargetSection(null);
-  }, []);
+  const combinedSideNavSections = useMemo(
+    () => [resourcesSection, ...navSections],
+    [resourcesSection, navSections]
+  );
+  const currentPageSection = useMemo(() => {
+    return combinedSideNavSections.find(
+      section => section.category === currentView?.category
+    );
+  }, [combinedSideNavSections, currentView]);
+
+  const resetExpandedSection = useCallback(
+    (closeAfterDelay = true) => {
+      if (stickyMode && currentPageSection) {
+        setPreviousExpandedSection(debouncedSection);
+        setTargetSection(currentPageSection);
+      } else {
+        setIsClosing(closeAfterDelay);
+        setPreviousExpandedSection(null);
+        setTargetSection(null);
+      }
+    },
+    [currentPageSection, stickyMode, debouncedSection]
+  );
+
+  useEffect(() => {
+    if (stickyMode) {
+      if (currentPageSection) {
```

Comment:
In what situation this is undefined?

Reply:
When you are on a page that isn't a sidenav section, (such as the Account Page). I'll add a comment.